ToolBox/CoinApi/CoinApiDataQueueHandler.cs

```diff
@@ -0,0 +1,360 @@
+                _subscribedSymbols = _subscribedSymbols.Where(x => !symbolsToUnsubscribe.Contains(x)).ToHashSet();
+
+                ProcessSubscriptionRequest();
+            }
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            _connectionHandler.DisposeSafely();
+
+            if (_webSocket.IsOpen)
+            {
+                _webSocket.Close();
+            }
+        }
+
+        /// <summary>
+        /// Helper method used in QC backend
+        /// </summary>
+        /// <param name="markets">List of LEAN markets (exchanges) to subscribe</param>
+        public void SubscribeMarkets(List<string> markets)
+        {
+            if (markets.Count == 0) return;
+
+            Log.Trace($"CoinApiDataQueueHandler.SubscribeMarkets(): {string.Join(",", markets)}");
+
+            _subscribedExchanges = markets.ToList();
+
+            var message = JsonConvert.SerializeObject(new HelloMessage
+            {
+                ApiKey = _apiKey,
+                Heartbeat = true,
+                SubscribeDataType = new[] { "trade", "quote" },
+                SubscribeFilterSymbolId = markets.Select(x => _symbolMapper.GetExchangeId(x)).ToArray()
+            });
+
+            _webSocket.Send(message);
+
+            _connectionHandler.EnableMonitoring(true);
+        }
+
+        private void ProcessSubscriptionRequest()
+        {
+            if (_subscriptionsPending) return;
+
+            _lastSubscribeRequestUtcTime = DateTime.UtcNow;
+            _subscriptionsPending = true;
+
+            Task.Run(() =>
+            {
+                while (true)
+                {
+                    DateTime requestTime;
+                    List<Symbol> symbolsToSubscribe;
+                    lock (_lockerSubscriptions)
+                    {
+                        requestTime = _lastSubscribeRequestUtcTime.Add(_subscribeDelay);
+                        symbolsToSubscribe = _subscribedSymbols.ToList();
+                    }
+
+                    if (DateTime.UtcNow > requestTime)
+                    {
+                        // minimum delay has passed since last subscribe request, send the subscribe request
+                        SubscribeSymbols(symbolsToSubscribe);
+
+                        lock (_lockerSubscriptions)
+                        {
+                            _lastSubscribeRequestUtcTime = DateTime.UtcNow;
+                            if (_subscribedSymbols.Count == symbolsToSubscribe.Count)
+                            {
+                                // no more subscriptions pending, task finished
+                                _subscriptionsPending = false;
+                                break;
+                            }
+                        }
+                    }
+
+                    Thread.Sleep(200);
```

Comment:
In general, we shouldn't be calling `Thread.Sleep` within a task object as it will block a thread pool thread. You can use the task constructs, such as `await Task.Delay(200).ConfigureAwait(false)` instead (will need to add `async` before the lambda.